DOCSP-45837 Adds Server Parameters style guidance (#183)
mongodb/docs-meta: source/style-guide/information-types/prototypes/reference/server-params-style.txt

.. _meta-reference-server-params:

=======================

> Server Parameters Style

Reviewer: [s] For consistency with the naming of other pages in the style guide, I would suggest: 
```suggestion
Server Parameters Guidelines
```

Author: Thanks for this suggestion. Updated!

=======================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

> 

Reviewer 2: [Suggestion]

It looks a bit odd to me to jump immediately from the title to this h1. I'm wondering if we can consider adding a contextual sentence before the h1 explaining the overall goal of this page, or maybe even remove the h1 heading entirely.

Author: Removed. Thanks!

When you document a new server parameter, include the following details 

> in addition to specific information about the new parameter. Documwent 

Author: Fixed. Thanks!

> When documenting a new server parameter, include the following details 
> in addition to specific information about the new parameter:

Reviewer 3: ```suggestion
When you document a new server parameter, include the following details 
in addition to specific information about the new parameter:
```

Author: Updated. Thanks!

> ought to be documented in the order they appear in the following table:

Reviewer 2: [Minor suggestion]

Consider addressing the writer here:

"Document these details in the order they appear in the following table: "

Author: Done. Thanks!

.. list-table::
   :widths: 40 40
   :header-rows: 1

>    * - Documentation Detail

Reviewer 2: [Question]

Are these details listed in the order that they should appear on the docs page? If so, can we make that clear?

Author: Updated. Thanks!

     - Necessity
     - Description

>    * - mongod / mongos Support

Reviewer 3: ```suggestion
   * - ``mongod`` or ``mongos`` Support
```

Author: Updated. Thanks!

     - Required
     - Whether the parameter is supported on ``mongod``, ``mongos``, or both.

>        Use ``|mongod-only|``, ``|mongos-only|``, or ``|both|`` as appropriate.

Reviewer 3: ```suggestion
       Use the ``|mongod-only|``, ``|mongos-only|``, or ``|both|`` substitutions as appropriate.
```

Author: Done.

   * - Version the Parameter was Added or Changed In

>      - Optional

Reviewer 3: Is this optional or conditional? Seems important when it applies.

Author: Conditional strikes me as more appropriate. Thanks!

       - For recent versions, include the version the parameter was added 
         or changed in.

>        - For archived versions, delete the version added tag.

Reviewer: [n] Not sure if list items are needed here, to me in a table it's a little jarring as the style is different from the other rows. I think each can just be a sentence on a new line within the cell.

Author: Good call. Updated!

       - For archived versions, delete the version added tag.

   * - OS Availability
     - Optional

>      - Some parameters are only available on certain operating systems. If the 
>        parameter is only supported on certain operating systems, specify which 
>        it's supported on.

Reviewer 3: ```suggestion
     - Some parameters are available only on certain operating systems. If the 
       parameter is supported only on certain operating systems, specify which 
       it's supported on.
```

Author: Fixed. Thanks!

   * - Argument Type
     - Required

>      - The type of argument the parameter expects: integer, string, boolean,
>        document, etc.

Reviewer 3: ```suggestion
     - Type of argument the parameter expects: integer, string, boolean,
       document, and so on.
```

Author: Fixed. Thanks!

   * - Default Value
     - Required

>      - Specify the default value the parameter takes.

Reviewer 3: ```suggestion
     - Specify the default value of the parameter.
```

Author: Done.

   
   * - Startup and Runtime Support
     - Required

>      - For details, see :ref:`startup-runtime-support`.

Reviewer 2: [Suggestion]

I might consider moving the content from line 63 to this table so we still retain a basic definition for this option in the table itself. Then, the "Startup and Runtime Support" section would begin with the line "You can use the provided includes to document startup / runtime support.", which I think works because we're getting into specific implementation details.

Author: Updated. Thanks!

.. _startup-runtime-support:

Startup and Runtime Support
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Whether the parameter is supported at startup, runtime, or both.

Reviewer 4: ```suggestion
Specify whether the parameter is supported at startup, runtime, or both.
```
Suggestion: Perhaps this should be a complete sentence since it's a new subsection

Author: Moved and updated. Thanks!

> You can use the provided includes to document startup / runtime support.

Reviewer 3: ```suggestion
You can use the provided includes to document startup or runtime support.
```

Author: Updated. Thanks!

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer 2: [Minor]

Since this is now an h1, let's use the hyphen underline `-` rather than tilde `~`. Same comment applies to the "Example" heading.

Author: Done. Thanks!

> To specify the parameter is only supported at startup, use:

Reviewer 3: ```suggestion
To specify that the parameter is supported only at startup, use the following include:
```

Author: Updated. Thanks!

.. code-block:: none

   .. include:: /includes/fact-startup-parameter  

> To specify the parameter is only supported at runtime, use:

Reviewer 3: ```suggestion
To specify that the parameter is supported only at runtime, use the following include:
```

Author: Updated. Thanks!

.. code-block:: none

   .. include:: /includes/fact-runtime-parameter  

> To specify the parameter is supported both at startup and during
> runtime, use:

Reviewer 3: ```suggestion
To specify the parameter is supported both at startup and during
runtime, use the following include:
```

Author: Updated. Thanks!

.. code-block:: none

   .. include:: /includes/fact-runtime-startup-parameter

Example
-------

The following screenshot shows the documentation for the 
``tlsUseSystemCA`` parameter. 

.. figure:: /images/server-param-example.png
   :alt: tlsUseSystemCA example

The following table describes the details that the ``tlsUseSystemCA``
section documents:

.. list-table::
   :widths: 30 30 40
   :header-rows: 1

   * - Documentation Detail
     - Necessity
     - Example in tlsUseSystemCA

>    * - mongod / mongos Support

Reviewer 3: ```suggestion
   * - ``mongod`` or ``mongos`` Support
```

Author: Updated. Thanks!

     - Required
     - Supported on ``mongod`` only

   * - Version the Parameter was Added or Changed In
     - Optional
     - Not applicable

   * - OS Availability
     - Optional
     - Not applicable

   * - Argument Type
     - Required
     - boolean

   * - Default Value
     - Required
     - false
   
   * - Startup and Runtime Support
     - Required

>      - ``tlsUseSystemCA`` is available at startup, but not runtime 

Reviewer 3: ```suggestion
     - ``tlsUseSystemCA`` is available at startup, but not runtime. 
```

Author: Done. Thanks!
